### PR TITLE
Added Draw State Functionality

### PR DESCRIPTION
**Pull request by stefprez**
Also created a Reflection Test Helper for testing private fields. The draw state has not been piped into the game at all yet. I'm not sure which class we'll be using for running the game.

### CODE REVIEW

File: test/model/ReflectionTestHelper.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        } catch (final SecurityException e) {
+            e.printStackTrace();
+        } catch (final IllegalArgumentException e) {
+            e.printStackTrace();
+        } catch (final IllegalAccessException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+}
```
Comment (csperlich) [inline test/model/ReflectionTestHelper.java]: Are the effects of field.setAccessible(true), felt after these methods are called? Should they be reset after the field is changed or extract?

Comment (stefprez) [inline test/model/ReflectionTestHelper.java]: I don't believe so. It's not actually changing the field to public, rather suppressing the Java language access checking for the field. More info [here](http://docs.oracle.com/javase/7/docs/api/java/lang/reflect/AccessibleObject.html#setAccessible%28boolean%29).

**Comment by csperlich** [comment]
About the liberal use of the keyword final? Is that something you'd like to start doing project wide or is that an eclipse auto-save setting you have, or is it both?

**Comment by stefprez** [comment]
It's an Eclipse auto-format on save setting! In general, it's better to mark variables final when possible for efficiency reasons, but I'm not sure how big of an impact it makes. I just know it's better that way than not using final, so I checked the box in the Eclipse auto-formatter.

**Pull request merged**